### DISCUSSION TITLE
How to properly switch bluetooth devices?

### DISCUSSION THREAD

**Discussion by Zombieschannel** [pschatzmann/ESP32-A2DP#436] [Q&A] [answered]
I am using a2dp_sink.
So from my testing, calling a2dp_sink.start() works perfectly. Unfortunately, I cannot figure out how to properly disconnect from the first device and connect to another one. 
I tried to use the following code: 
```
a2dp_sink->set_auto_reconnect(0);
a2dp_sink->disconnect();
a2dp_sink->set_discoverability(ESP_BT_GENERAL_DISCOVERABLE);
```
Though it seems like after disconnecting, I just get some sound coming out of the speaker and once I connect to a new device, the esp32 completely crashes/restarts.
The best solution I have currently is to just disable auto reconnecting and restart the esp32 and connect with a different device.

**Comment by pschatzmann** [comment, marked as answer]
When I was testing, I was disconnecting from my phone and then reconnecting from the same or another device...
I confirm that you better set auto_reconnect(false);

If you don't disconnect from the phone you run the risk that the phone might try to reconnect which might mess up what you try to do
Did you already try to call end() and then start() ?

I don't think that you should call set_discoverability() yourself, but rely on the API to set this automatically based on the status.

**Reply by Zombieschannel** [reply]
I got it to work quite well by calling set_auto_reconnect(0) at the start of the program instead of when disconnecting, also just before disconnecting set_volume(0); makes the "speaker noise sound" go away.

Edit: The end() start() works!

**Reply by Zombieschannel** [reply]
Thanks for the quick response + solution!
